reinforcement_learning/bindings/python/py_api.cc

```diff
@@ -0,0 +1,117 @@
+      reinforcement_learning::api_status status;
+      reinforcement_learning::utility::config::create_from_json(config_json, config, &status);
+      check_api_status(status);
+
+      return config;
+    }
+
+    live_model::live_model(const reinforcement_learning::utility::configuration config, error_callback& callback)
+      : impl(config, &dispatch_error_internal, &callback, &data_transport_factory, &model_factory)
+    {}
+
+    live_model::live_model(const reinforcement_learning::utility::configuration config)
+      : impl(config, nullptr, nullptr, &data_transport_factory, &model_factory)
+    {}
+
+    void live_model::init() {
+      reinforcement_learning::api_status status;
+      impl.init(&status);
+      check_api_status(status);
+    }
+
+    ranking_response convert_ranking_response(const reinforcement_learning::ranking_response& response_impl) {
+      ranking_response response;
+      reinforcement_learning::api_status status;
+
+      response.event_id = response_impl.get_event_id();
+      response.model_id = response_impl.get_model_id();
+
+      size_t chosen_action_id;
+      response_impl.get_chosen_action_id(chosen_action_id, &status);
+      check_api_status(status);
+      response.chosen_action_id = chosen_action_id;
+
+      std::vector<int> action_ids;
+      std::vector<float> possibilities;
+      action_ids.reserve(response_impl.size());
+      possibilities.reserve(response_impl.size());
+
+      for (const auto& action_prob : response_impl) {
+        action_ids.push_back(action_prob.action_id);
+        possibilities.push_back(action_prob.probability);
+      }
+
+      response.action_ids = std::move(action_ids);
+      response.probabilities = std::move(possibilities);
+
+      return response;
+    }
+
+    ranking_response live_model::choose_rank(const char* event_id, const char* context_json) {
+      reinforcement_learning::ranking_response response_impl;
+      reinforcement_learning::api_status status;
+      impl.choose_rank(event_id, context_json, response_impl, &status);
+      check_api_status(status);
+
+      return convert_ranking_response(response_impl);
+    }
+    // event_id is auto-generated.
+    ranking_response live_model::choose_rank(const char* context_json) {
+      reinforcement_learning::ranking_response response_impl;
+      reinforcement_learning::api_status status;
+      impl.choose_rank(context_json, response_impl, &status);
+      check_api_status(status);
+
+      return convert_ranking_response(response_impl);
+    }
+
+    void live_model::report_outcome(const char* event_id, const char* outcome) {
+      reinforcement_learning::api_status status;
+      impl.report_outcome(event_id, outcome, &status);
+      check_api_status(status);
+    }
+
+    void live_model::report_outcome(const char* event_id, float outcome) {
+      reinforcement_learning::api_status status;
+      impl.report_outcome(event_id, outcome, &status);
+      check_api_status(status);
+    }
+  }
+}
```

Comment:
[minor] missing new line to be POSIX compliant

Reply:
✔️